src/problem.jl

```diff
@@ -254,22 +254,46 @@ end
 function extend_problem(prob::DiffEqBase.AbstractODEProblem, jumps; rng = DEFAULT_RNG)
     _f = SciMLBase.unwrapped_f(prob.f)
 
-    jump_f = let _f = _f
-        function jump_f(du::ExtendedJumpArray, u::ExtendedJumpArray, p, t)
-            _f(du.u, u.u, p, t)
-            update_jumps!(du, u, p, t, length(u.u), jumps...)
+    if isinplace(prob)
+        jump_f = let _f = _f
+            function (du::ExtendedJumpArray, u::ExtendedJumpArray, p, t)
+                _f(du.u, u.u, p, t)
+                update_jumps!(du, u, p, t, length(u.u), jumps.variable_jumps...)
```

Comment:
`extend_problem` is now passed just the variable jumps from `JumpProblem`, and not the full jump set since that is all that it was using. I did this because we now filter the types of variable rate jumps from `jumps.variable_jumps`, so only a subset of them may be wanted / handled here.

Reply:
Otherwise `JumpProblem` would need to create a new `JumpSet` that just holds the new (subset) of `VariableRateJump`s needed here.

Reply:
I see, it's because I was working off an old master branch, rebasing makes all of the failures go away except in my code.